src/clib/pio_nc.c

```diff
@@ -847,9 +847,9 @@ int PIOc_inq_var(int ncid, int varid, char *name, nc_type *xtypep, int *ndimsp,
             }
             if (!ierr)
                 ierr = ncmpi_inq_var(file->fh, varid, my_name, xtypep, ndimsp, (dimidsp)?dimidsp:tmp_dimidsp, nattsp);
-            if (!ierr && name)
+            if (!ierr && name && max_namelen > 0)
             {
-                strncpy(name, my_name, PIO_MAX_NAME);
+                strncpy(name, my_name, (max_namelen < PIO_MAX_NAME) ? max_namelen : PIO_MAX_NAME);
```

Comment:
Instead of this logic it would be better to have an assert to check that namelen <= PIO_MAX_NAME+1, change the size of cached name to PIO_MAX_NAME+1, and change the name of the arg from max_namelen to just namelen.

Reply:
Thanks, I have made the code changes and rebased the feature branch.